src/ray/core_worker/reference_count.h

```diff
@@ -820,6 +823,10 @@ class ReferenceCounter : public ReferenceCounterInterface,
   /// locally.
   absl::flat_hash_set<ObjectID> freed_objects_ GUARDED_BY(mutex_);
 
+  /// Index that all local reference counts to speed up the process of releasing
+  /// all local references when shutdown.
+  absl::flat_hash_map<ObjectID, int32_t> local_ref_counts_ GUARDED_BY(mutex_);
```

Comment:
Since shutdown is not performance critical, how about not indexing this and instead scanning the entire reference table for local references?